### PR TITLE
Update to abseil-cpp source tag 20211102.0

### PR DESCRIPTION
**Pull request by dennycd**
### Base Source Import  

* Abseil-cpp source snapshot import from latest release tag [20211102.0 ](https://github.com/abseil/abseil-cpp/releases/tag/20211102.0)
     -  Upstream [grpc c-core dependency ](https://github.com/grpc/grpc/blob/master/bazel/grpc_deps.bzl#L285-L294)

### Additional Change 

* Package.swift SPM update for the 20211102.0 source tree change
* Pragma workaround for nested namespace error (grafted from [paulb777/abseil-cpp](https://github.com/paulb777/abseil-cpp/commit/6a901b311ba479b388f517f6aa59429c5e48e6f0))

### Build and Test Verification 

```
swift build --target abseil
swift test
```

cc @paulb777  @wu-hui

https://github.com/grpc/grpc-ios/issues/47

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/00-bug_report.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+name: Bug report
```
Comment (paulb777) [inline .github/ISSUE_TEMPLATE/00-bug_report.md]: The Issue template files should not be added. Otherwise, people will do the wrong things with bugs here.

Comment (dennycd) [reply .github/ISSUE_TEMPLATE/00-bug_report.md]: removed

File: Package.swift [hunk 2]
```diff
@@ -24,6 +24,28 @@ let package = Package(
         "absl/strings/str_format_test.cc",
         "absl/strings/ascii_test.cc",
         "absl/strings/escaping_test.cc",
+        "absl/strings/match_test.cc",
```
Comment (paulb777) [inline Package.swift]: Would it be better to not mirror the test.cc and benchmark.cc files into the repo in the first place, to make the Package.swift simpler and avoid having to keep adding new ones on an ongoing basis?

**Review by paulb777 — APPROVED** [review]

File: Package.swift (hunk 2 above)
Comment (dennycd) [reply Package.swift]: nice, I was able to batch remove most of the testing & benchmark files. Updated

**Pull request merged**